compliance_checker/cf/cf_base.py

```diff
@@ -495,11 +495,12 @@ def _find_aux_coord_vars(self, ds, refresh=False):
         :return: List of variable names (str) that are defined to be auxiliary
                  coordinate variables.
         """
+        ds_str = ds.__str__()
```

Comment:
@benjwadams this is not elegant but using the actual `netCDF4.Dataset` object here was leading to odd behavior.  I guess that the str representation of the dataset is unique and "safe enough" to use here. Maybe a hash would be more appropriated but it tuns hashing netCDF objects is not easy!